pkgs/development/libraries/librime/default.nix

```diff
@@ -20,12 +20,26 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-tflWBfH1+1AFvkq0A6mgsKl+jq6m5c83GA56LWxdnlw=";
   };
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ cmake pkg-config xorgproto utf8cpp ];
 
   buildInputs = [ boost glog leveldb marisa opencc yaml-cpp gtest capnproto ]
               ++ plugins; # for propagated build inputs
 
-  preConfigure = copyPlugins;
+  strictDeps = true;
+
+  postPatch = ''
+    rm --recursive --verbose include/{utf8,utf8.h,X11}
+  '';
+
+
+  preConfigure = ''
+    cmakeFlagsArray=(
+      $cmakeFlagsArray
+      "-DCMAKE_CXX_FLAGS=-I${lib.getDev utf8cpp}/include/utf8cpp -I${lib.getDev xorgproto}/include"
```

Comment:
Whitespaces still don't work in `cmakeFlags`?

Reply:
Right. `cmakeFlags = [ "-DCMAKE_CXX_FLAGS='-I... -I...'" ]` is parsed to `"-DCMAKE_CXX_FLAGS=-I..." "-I..."`in bash for me.